pandas/core/dtypes/cast.py

```diff
@@ -246,6 +246,97 @@ def trans(x):
     return result
 
 
+def maybe_cast_result(
+    result, obj: ABCSeries, numeric_only: bool = False, how: str = ""
+):
+    """
+    Try casting result to a different type if appropriate
+
+    Parameters
+    ----------
+    result : array-like
```

Comment:
Does "array-like" in this context mean "ndarray or EA" or does it include Index/Series/??.  If its just the first two, then we can annotate `result` as `ArrayLike`

Reply:
I added the line:

    assert is_extension_array_dtype(result) or isinstance(result, np.ndarray)

to this function and ran the tests and got 7 failures. One of which was 

    tests.resample.test_datetime_index.test_resample_median_bug_1688

where the type is Series.